summarisation_map_reduce.py

```diff
@@ -0,0 +1,56 @@
+from langchain.chains.mapreduce import MapReduceChain
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.chains import ReduceDocumentsChain, MapReduceDocumentsChain
+
+llm = ChatOpenAI(temperature=0)
+
+# Map
+map_template = """The following is a set of documents
+{docs}
+Based on this list of docs, please identify the main themes 
+Helpful Answer:"""
+map_prompt = PromptTemplate.from_template(map_template)
+map_chain = LLMChain(llm=llm, prompt=map_prompt)
+
+# Reduce
+reduce_template = """The following is set of summaries:
+{doc_summaries}
+Take these and distill it into a final, consolidated summary of the main themes. 
+Helpful Answer:"""
+reduce_prompt = PromptTemplate.from_template(reduce_template)
+reduce_chain = LLMChain(llm=llm, prompt=reduce_prompt)
+
+# Takes a list of documents, combines them into a single string, and passes this to an LLMChain
+combine_documents_chain = StuffDocumentsChain(
+    llm_chain=reduce_chain, document_variable_name="doc_summaries"
+)
+
+# Combines and iteravely reduces the mapped documents
+reduce_documents_chain = ReduceDocumentsChain(
+    # This is final chain that is called.
+    combine_documents_chain=combine_documents_chain,
+    # If documents exceed context for `StuffDocumentsChain`
+    collapse_documents_chain=combine_documents_chain,
+    # The maximum number of tokens to group documents into.
+    token_max=4000,
+)
+
+# Combining documents by mapping a chain over them, then combining results
+map_reduce_chain = MapReduceDocumentsChain(
+    # Map chain
+    llm_chain=map_chain,
+    # Reduce chain
+    reduce_documents_chain=reduce_documents_chain,
+    # The variable name in the llm_chain to put the documents in
+    document_variable_name="docs",
+    # Return the results of the map steps in the output
+    return_intermediate_steps=False,
+)
+
+text_splitter = CharacterTextSplitter.from_tiktoken_encoder(
+    chunk_size=1000, chunk_overlap=0
+)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `CharacterTextSplitter` class and `docs` variable are used but not defined in this context. This could lead to a `NameError`. Please ensure that these are properly defined or imported before use.
```diff
+ docs = ["Document 1", "Document 2", "Document 3"] # example documents
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->